File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const isVisited = Array(n).fill(false);
    
    
    for (let index = 0; index < n; index++) {
        if (!isVisited[index]) {
            answer++;
            dfs(index);
        }
    }
    
    function dfs (index) {
        isVisited[index] = true;
        for (let i = 0; i < n; i++) {
            const isConnected = computers[index][i] === 1 ? true : false;
            
            if (isConnected && !isVisited[i]) {
                dfs(i);
            }
        }
        
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def dfs(computer_index):
        visited[computer_index] = True
        
        for i in range(n):
            if computers[computer_index][i] == 1 and not visited[i]:
                dfs(i)
                
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
    return answer